src/test/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/RaptorCostLinearFunctionTest.java

```diff
@@ -0,0 +1,90 @@
+package org.opentripplanner.routing.algorithm.raptoradapter.transit.cost;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.framework.model.Cost;
+import org.opentripplanner.routing.api.request.framework.CostLinearFunction;
+import org.opentripplanner.test.support.TestTableParser;
+
+class RaptorCostLinearFunctionTest {
+
+  private static final double COEFFICIENT = 2.5;
+  private static final int CONSTANT_SECONDS = 17;
+
+  private final CostLinearFunction sameInOtpDomain = CostLinearFunction.of(
+    Cost.costOfSeconds(CONSTANT_SECONDS),
+    COEFFICIENT
+  );
+
+  private final RaptorCostLinearFunction subject = RaptorCostLinearFunction.of(
+    CONSTANT_SECONDS,
+    COEFFICIENT
+  );
+
+  static Stream<Arguments> calculateRaptorCostTestCases() {
```

Comment:
Now that I see this I would like to point out that I;ve recently discovered a new way to write parameterized tests with Junit 5: DynamicTests.

Example: https://github.com/opentripplanner/OpenTripPlanner/blob/d36c0f0b7738693d6fb07b5b6531974385a8744d/src/ext-test/java/org/opentripplanner/ext/flex/AreaStopsToVerticesMapperTest.java#L54

It's not as compact as I would like but it does cut down on a few untyped, string references.

Not sure if it would help in this instance.

Reply:
I do not think this code is easier to read/understand than using a @PerameterizedTest
```
return testCases
      .stream()
      .map(tc ->
        dynamicTest(
          tc.toString(),
          () -> {
```
I can see that @TestFactory is helpful if you write a test framework on top of JUnit, but I doubt that this is the intended use for it.

Reply:
We talked about this at the OTP meeting today, and the conclusion is that we want to use the `@ParameterizedTest` annotation and not switch to test factory.